### PR TITLE
feat: Compile the codersdk for wasm

### PR DESCRIPTION
**Pull request by kylecarbs**
```
GOOS=js GOARCH=wasm go build -o main.wasm ./codersdk
```
This enables the sdk to be built for WebAssembly. It also adds a few functions to the tailnet conn that will be used for integration into a VS Code extension.

### CODE REVIEW

File: codersdk/provisionerdaemons.go [hunk 1]
```diff
@@ -130,10 +130,7 @@ func (c *Client) provisionerJobLogsAfter(ctx context.Context, path string, after
 	httpClient := &http.Client{
 		Jar: jar,
 	}
-	conn, res, err := websocket.Dial(ctx, followURL.String(), &websocket.DialOptions{
-		HTTPClient:      httpClient,
-		CompressionMode: websocket.CompressionDisabled,
-	})
+	conn, res, err := websocket.Dial(ctx, followURL.String(), websocketOptions(httpClient, 0))
```
Comment (mafredri) [inline codersdk/provisionerdaemons.go:133]: Should we use `websocket.CompressionNoContextTakeover` instead of 0 for these?

File: tailnet/conn.go [hunk 2]
```diff
@@ -557,6 +555,31 @@ func (c *Conn) sendNode() {
...
+	node := &Node{
+		ID:            c.netMap.SelfNode.ID,
+		AsOf:          database.Now(),
+		Key:           c.netMap.SelfNode.Key,
+		Addresses:     c.netMap.SelfNode.Addresses,
+		AllowedIPs:    c.netMap.SelfNode.AllowedIPs,
+		DiscoKey:      c.magicConn.DiscoPublicKey(),
+		Endpoints:     c.lastEndpoints,
+		PreferredDERP: c.lastPreferredDERP,
+		DERPLatency:   c.lastDERPLatency,
```
Comment (mafredri) [inline tailnet/conn.go:575]: Do we need to hold a mutex to access any of these values? Maybe this function should only be called when a mutex is held? Should we comment that?

**Review by mafredri — APPROVED** [review]
I'm not sure if you're still planning on merging this, but LGTM!

**Comment by kylecarbs** [comment]
@mafredri I'll leave it for the archives! I'd rather not have this abstraction over our WebSocket code, so if we have a WASM use-case in the future it can exist for that.

**Pull request closed without merging**